modules/olm-disconnected-operatorhub-disconnected-mode.adoc

```diff
@@ -0,0 +1,151 @@
+= Configuring OperatorHub in disconnected mode
+
+Cluster administrators can configure OLM and OperatorHub to use local content in
+a disconnected mode.
+
+.Prerequisites
+
+* Cluster administrator access to a connected or disconnected {product-title} cluster and its internal registry.
+* Separate workstation with full Internet access.
+
+.Procedure
+
+. **Disable the default OperatorSources.**
+
+.. Edit the `OperatorHub` configuration resource:
++
+----
+$ oc edit OperatorHub cluster
+----
+
+.. Add `disableAllDefaultSources: true` to the spec:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: OperatorHub
+spec:
+  disableAllDefaultSources: true
+----
++
+This disables the default OperatorSources that are configured by default during
+a {product-title} installation.
+
+. **Retrieve package lists.**
++
+To get the list of packages that are available for the default OperatorSources,
+run the following `curl` commands:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages?namespace=redhat-operators > packages.txt
+$ curl https://quay.io/cnr/api/v1/packages?namespace=community-operators >> packages.txt
+$ curl https://quay.io/cnr/api/v1/packages?namespace=certified-operators >> packages.txt
+----
++
+Each package in the new `packages.txt` is an Operator that you could add to your
+disconnected catalog. From this list, you could either pull every Operator or a
+subset that you would like to expose to users.
+
+. **Pull Operator content.**
++
+For a given Operator in the package list, you must pull its content. This
+example uses the `openshift-template-service-broker` Operator:
+
+.. Retrieve the digest:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages/redhat-operators/openshifttemplateservicebroker/3.0.0
+----
+
+.. From that JSON, take the digest and use it to pull the gzipped archive:
++
+----
+$ curl -XGET https://quay.io/cnr/api/v1/packages/redhat-operators/openshifttemplateservicebroker/blobs/sha256/9bddc82c0fee05b243d01c6e2f0110c413eb9165cd7bb8adc25f1e88997854c9 -o openshifttemplateservicebroker.tar.gz
+----
+
+.. To pull the information out, you must untar the archive into a directory with
+all the other Operators that you want. For example, to untar to a directory
+called `manifests/`:
++
+----
+$ tar fczP openshifttemplateservicebroker.tar.gz manifests/
+----
+
+.. In this directory, you must now handle two different types of bundles:
+
+... If you see a single file called `bundle.yaml`, you must break `data.clusterServiceVersion` (each file in the list), `data.Package`, and `data.customResourceDefinition` (each file in the list) into their own files:
++
+----
+manifests/
+└── openshifttemplateservicebroker
+      ├── clusterserviceversion.yaml
+      ├── customresourcedefinition.yaml
+      ├── package.yaml
+└── etcd-<version>
+  └─ <CSVs_and_CRDs_and_a_package_file>
+----
+
+... If you see files already in the structure described in the previous step, then you do not have to do anything.
```

Comment:
Shouldn't they be in this format (based on the instructions we are giving in the piro step?

Reply:
No, there are two different things in the tar potentially. one that is just `bundle.yaml` and one that is broken up already. This is stating that if you don't see the bundle.yaml then you don't need to do anything. Can we make that more clear? Does that make sense?